### PR TITLE
Use configured temporary directory for recovering snapshots

### PR DESCRIPTION
**Pull request by timvisee**
Fixes <https://github.com/qdrant/qdrant/issues/2127>.

### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### CODE REVIEW

**Review by agourlay — APPROVED** [review]
Code looks good :+1: 

Were you able to test this manually?

**Comment by timvisee** [comment]
> Were you able to test this manually?

Yes, I did test it manually :smile: 

It is non-trivial to automate testing for this change specifically, but the relevant functions are already called by other tests.

File: src/snapshots.rs [hunk 1]
```diff
@@ -78,23 +82,26 @@ pub fn recover_snapshots(
...
     snapshot_path: &str,
     storage_dir: &str,
     force: bool,
     this_peer_id: PeerId,
     is_distributed: bool,
 ) -> Vec<String> {
-    let temporary_dir = Path::new(storage_dir).join("snapshots_recovery_tmp");
-    std::fs::create_dir_all(&temporary_dir).unwrap();
+    let snapshot_temp_path = temp_dir.map(PathBuf::from).unwrap_or_else(|| {
+        Path::new(storage_dir).join(format!("snapshots-recovery-{}", Uuid::new_v4()))
```
Comment (coszio) [inline src/snapshots.rs:93]: Why do we need a uuid here?

Comment (timvisee) [reply src/snapshots.rs]: Indeed not needed because it is called once. Reverted it to how it was before. Thanks!

**Review by coszio — APPROVED** [review]

**Pull request merged**